contracts/RewardsManager.vy

```diff
@@ -1,51 +1,173 @@
-# @version 0.3.0
-# @notice A manager contract for the StakingRewards contract.
-# @author skozin
+# @version 0.3.1
+# @notice Wrapper contract for the Balancer Liquidity Gauge.
+# @author bulbozaur <info@lido.fi>
 # @license MIT
+
 from vyper.interfaces import ERC20
 
 
-interface StakingRewards:
-    def periodFinish() -> uint256: view
-    def notifyRewardAmount(reward: uint256, rewardHolder: address): nonpayable
+struct BalancerReward:
+    token: address
+    distributor: address
+    period_finish: uint256
+    rate: uint256
+    last_update: uint256
+    integral: uint256
+
+
+interface BalancerLiquidityGauge:
+    def reward_data(addr: address) -> BalancerReward: view
+    def deposit_reward_token(_reward_token: address, _amount: uint256): nonpayable
+    def set_reward_distributor(_reward_token: address, _distributor: address): nonpayable
+
+
+event OwnershipTransferred:
+    previousOwner: indexed(address)
+    newOwner: indexed(address)
+
+
+event RewardsContractUpdated:
+    newRewardsContract: indexed(address)
+
+
+event RewardsContractTransferred:
+    newDistributor: indexed(address)
+
+
+event WeeklyRewardsAmountUpdated:
```

Comment:
There are no tests for events `NewRewardsPeriodStarted` and `RewardsContractTransferred`

Reply:
RewardsContractTransferred has one, added for NewRewardsPeriodStarted